From b8c5c996e888485c3a16d645c8490592534a7882 Mon Sep 17 00:00:00 2001
From: Jesse Barnes <jbarnes@virtuousgeek.org>
Date: Tue, 15 Sep 2009 19:45:47 -0700
Subject: [PATCH 2525/2543] Eliminate cursor flicker

We can update the cursor without hiding and showing it.  In fact, doing the
hide/show causes noticable flicker when running in KMS mode.

Signed-off-by: Jesse Barnes <jbarnes@virtuousgeek.org>
---
 src/i830_cursor.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/i830_cursor.c b/src/i830_cursor.c
index a6aba41..83af35f 100644
--- a/src/i830_cursor.c
+++ b/src/i830_cursor.c
@@ -119,6 +119,7 @@ I830CursorInit(ScreenPtr pScreen)
 			       HARDWARE_CURSOR_SWAP_SOURCE_AND_MASK |
 			       HARDWARE_CURSOR_AND_SOURCE_WITH_MASK |
 			       HARDWARE_CURSOR_SOURCE_MASK_INTERLEAVE_64 |
+			       HARDWARE_CURSOR_UPDATE_UNHIDDEN |
 			       HARDWARE_CURSOR_ARGB));
 }
 
-- 
1.6.3.3

